pkg/cmd/kafka/create/create.go

```diff
@@ -700,35 +723,107 @@ func promptKafkaPayload(opts *options, constants *remote.DynamicServiceConstants
+			Options: validMarketPlaces,
 		}
-		// casting to nullable string
-		var clusterIdSNS kafkamgmtclient.NullableString
-		clusterIdSNS.Set(opts.selectedCluster.ClusterId)
+		err = survey.AskOne(marketplacePrompt, &answers.Marketplace)
+		if err != nil {
+			return nil, err
+		}
+	}
 
-		var billingModelEnterprise kafkamgmtclient.NullableString
-		var enterprise = "enterprise"
-		billingModelEnterprise.Set(&enterprise)
+	if len(validMarketPlaces) > 0 {
 
-		var nullString kafkamgmtclient.NullableString
-		payload = &kafkamgmtclient.KafkaRequestPayload{
-			Name:                  answers.Name,
-			BillingModel:          billingModelEnterprise,
-			BillingCloudAccountId: nullString,
-			Marketplace:           nullString,
-			ClusterId:             clusterIdSNS,
-			//	set plan here
+		validMarketplaceAcctIDs := FetchValidMarketplaceAccounts(orgQuota.MarketplaceQuotas, answers.Marketplace)
+
+		if len(validMarketplaceAcctIDs) == 1 {
+			answers.MarketplaceAcctID = validMarketplaceAcctIDs[0]
+		} else {
+			marketplaceAccountPrompt := &survey.Select{
+				Message: f.Localizer.MustLocalize("kafka.create.input.marketplaceAccountID.message"),
+				Options: validMarketplaceAcctIDs,
+			}
+			err = survey.AskOne(marketplaceAccountPrompt, &answers.MarketplaceAcctID)
+			if err != nil {
+				return nil, err
+			}
 		}
 	}
-	return payload, nil
+	return answers, nil
+}
+
+func cloudProviderPrompt(f *factory.Factory, answers *promptAnswers) (*promptAnswers, error) {
+	cloudProviderNames, err := GetEnabledCloudProviderNames(f)
+	if err != nil {
+		return nil, err
+	}
+
+	cloudProviderPrompt := &survey.Select{
+		Message: f.Localizer.MustLocalize("kafka.create.input.cloudProvider.message"),
+		Options: cloudProviderNames,
+	}
+
+	err = survey.AskOne(cloudProviderPrompt, &answers.CloudProvider)
+	if err != nil {
+		return nil, err
+	}
+	return answers, nil
+}
+
+// This may need to altered as the `name` are mutable on ocm side
+func getClusterNameMap(f *factory.Factory, clusterList *kafkamgmtclient.EnterpriseClusterList) (*map[string]v1.Cluster, error) {
+	//	for each cluster in the list, get the name from ocm and add it to the cluster list
+	str := listclusters.CreateSearchString(clusterList)
+	ocmClusterList, err := clustermgmt.GetClusterListByIds(f, "", "", str, len(clusterList.Items))
+	if err != nil {
+		return nil, err
+	}
+	clusterMap := make(map[string]v1.Cluster)
+	for _, cluster := range ocmClusterList.Slice() {
+		clusterMap[cluster.ID()] = *cluster
+	}
+	return &clusterMap, nil
+
+}
+
+func validateClusters(clusterList *kafkamgmtclient.EnterpriseClusterList) *kafkamgmtclient.EnterpriseClusterList {
+	// if cluster is in a ready state add it to the list of clusters
+	returnList := kafkamgmtclient.EnterpriseClusterList{}
+	for _, cluster := range clusterList.Items {
+		if *cluster.Status == "ready" {
+			returnList.Items = append(returnList.Items, cluster)
+		}
+	}
+	// TO-DO add logic to return cluster capacity
```

Comment:
Should we target this TODO in the context of this PR?

Reply:
this was a question I wanted to ask a little more broadly, to determine capacity for each cluster we've got to do a seperate call to ocm for each cluster, to the best of my knowledge. 
is this overkill? I mean I know it would be nice to see this info here but I was just wondering?